components/common/Modal/modal.tsx

```diff
@@ -17,13 +17,11 @@ const Modal = ({ children, show, ...props }: IModalProps) => {
   const isDesktop = useIsDesktop()
 
   useEffect(() => {
-    if (show) {
-      document.body.style.overflow = 'hidden'
-      document.body.style.touchAction = 'none'
-    } else {
-      document.body.style.overflow = 'unset'
-      document.body.style.touchAction = 'auto'
-    }
+    const TopScroll = window.pageYOffset || document.documentElement.scrollTop
+    const LeftScroll = window.pageXOffset || document.documentElement.scrollLeft
+
+    if (show) window.onscroll = () => window.scrollTo(LeftScroll, TopScroll)
+    else window.onscroll = () => null
```

Comment:
makes scroll less visibly buggy and maybe move it to `addEventListener` to not block anything in future just in case ?
```suggestion
    const topScroll = window.pageYOffset || document.documentElement.scrollTop
    const leftScroll = window.pageXOffset || document.documentElement.scrollLeft

    if (show) window.onscroll = () => window.scrollTo({
          top: topScroll,
          left: leftScroll,
          behavior: 'instant',
        })
    else window.onscroll = () => null
```